### PR TITLE
Refactor: standardize epoch settings naming

### PR DESCRIPTION
**Pull request by dlachaume**
## Content

This PR includes refactoring of epoch settings:
- Moved `EpochSettings` from common to `SignerEpochSettings` in the signer
- Standardized occurrences of `EpochSetting` to `EpochSettings`
- Renamed the `ProtocolParametersStorer` trait to `EpochSettingsStorer`

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

### CODE REVIEW

File: mithril-aggregator/src/dependency_injection/builder.rs [hunk 1]
```diff
@@ -554,10 +554,8 @@ impl DependenciesBuilder {
         Ok(self.verification_key_store.as_ref().cloned().unwrap())
     }
 
-    async fn build_protocol_parameters_store(
-        &mut self,
-    ) -> Result<Arc<dyn ProtocolParametersStorer>> {
-        let protocol_parameters_store = EpochSettingStore::new(
+    async fn build_protocol_parameters_store(&mut self) -> Result<Arc<dyn EpochSettingsStorer>> {
```
Comment (Alenar) [inline mithril-aggregator/src/dependency_injection/builder.rs:557]: Little oversight :)
```suggestion
    async fn build_epoch_settings_storer(&mut self) -> Result<Arc<dyn EpochSettingsStorer>> {
```

**Review by Alenar — APPROVED** [review]
LGTM

File: mithril-aggregator/src/dependency_injection/builder.rs (hunk 1 above)
Comment (dlachaume) [reply mithril-aggregator/src/dependency_injection/builder.rs]: Done ✅

File: mithril-aggregator/src/dependency_injection/builder.rs [hunk 2]
```diff
@@ -554,10 +554,8 @@ impl DependenciesBuilder {
...
+        let protocol_parameters_store = EpochSettingsStore::new(
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-aggregator/src/dependency_injection/builder.rs]: Done ✅

File: mithril-aggregator/src/dependency_injection/builder.rs [hunk 3]
```diff
@@ -1305,7 +1301,7 @@ impl DependenciesBuilder {
             certificate_repository: self.get_certificate_repository().await?,
             open_message_repository: self.get_open_message_repository().await?,
             verification_key_store: self.get_verification_key_store().await?,
-            protocol_parameters_store: self.get_protocol_parameters_store().await?,
+            protocol_parameters_store: self.get_epoch_settings_storer().await?,
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-aggregator/src/dependency_injection/builder.rs]: Done ✅

File: mithril-aggregator/src/dependency_injection/containers.rs [hunk 4]
```diff
@@ -80,7 +79,7 @@ pub struct DependencyContainer {
     pub verification_key_store: Arc<dyn VerificationKeyStorer>,
 
     /// Protocol parameter store.
-    pub protocol_parameters_store: Arc<dyn ProtocolParametersStorer>,
+    pub protocol_parameters_store: Arc<dyn EpochSettingsStorer>,
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-aggregator/src/dependency_injection/containers.rs]: Done ✅

File: mithril-aggregator/src/services/epoch_service.rs [hunk 5]
```diff
@@ -107,15 +107,15 @@ pub struct MithrilEpochService {
     future_protocol_parameters: ProtocolParameters,
     epoch_data: Option<EpochData>,
     computed_epoch_data: Option<ComputedEpochData>,
-    protocol_parameters_store: Arc<dyn ProtocolParametersStorer>,
+    protocol_parameters_store: Arc<dyn EpochSettingsStorer>,
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-aggregator/src/services/epoch_service.rs]: Done ✅

File: mithril-aggregator/src/services/epoch_service.rs [hunk 6]
```diff
@@ -617,7 +617,7 @@ mod tests {
             }
         }
 
-        let protocol_parameters_store = FakeProtocolParametersStorer::new(vec![
+        let protocol_parameters_store = FakeEpochSettingsStorer::new(vec![
```
[the remark this answers is not in the document]
Comment (dlachaume) [reply mithril-aggregator/src/services/epoch_service.rs]: Done ✅

**Review by Alenar — APPROVED** [review]

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Pull request merged**